### PR TITLE
Track One Event Hubs Tests Infrastructure, Processor and Client Live Tests Updates

### PR DESCRIPTION
**Pull request by tzhanl**
Track One Event Hubs Tests Infrastructure Updates:
Port EventHubScope,
Update TestConstant and TestUtility
Include more references to test project file
resolve namespace conflict introduce by Microsoft.Bcl.AsyncInterfaces
Processor Live Tests
Client Live Tests

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Infrastructure/EventHubScope.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Infrastructure/EventHubScope.cs]: This really doesn't belong hanging on top of the document comments; please remove.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Infrastructure/TestConstants.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Infrastructure/TestConstants.cs]: This isn't relevant to future users/maintainers of this code; please remove.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Infrastructure/TestConstants.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Infrastructure/TestConstants.cs]: Agree this would be unused; please remove instead of commenting out.   They can always be restored from source control history if need be.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Infrastructure/TestUtility.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Infrastructure/TestUtility.cs]: Please remove all references to copying code from elsewhere; it's not relevant for context of future maintainers.

**Comment by jsquire** [comment]
Please adapt the code that was copied for consistency with the local idioms and styles.  Right now, the copied code has full doc comments and the existing code does not; for conformity with the track one code, we should remove these.    Our goal is for the new code to blend with the existing code as much as possible rather than keep to its own style.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorNegativeCases.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorNegativeCases.cs]: I don't see these comments as adding value for future maintainers;  please remove.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTestBase.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTestBase.cs]: Please remove unused code blocks rather than commenting.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTestBase.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTestBase.cs]: nit:  please use the lowercase `string[]`

Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTestBase.cs]: Not sure why this intermediate variable is needed.   Couldn't we simply return `eventHubInfo.PartitionIds`?

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTestBase.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTestBase.cs]: I may be missing something, but does this need to be a `[LiveTest]`?  I don't see it actually using the scope at all, just testing local artifacts.   If not, let's please remove the attribute and scope so this runs as a standard unit test.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTestBase.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTestBase.cs]: nit: extra indenting

**Review by jsquire — CHANGES_REQUESTED** [review]
Looks good, overall.   I made some requests, mostly related to style and formatting.   I have not yet pulled down the set and run it locally to validate; once we're settled on everything else, I'll use that as my last approval gate.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Infrastructure/EventHubScope.cs
Comment (tzhanl) [reply sdk/eventhub/Microsoft.Azure.EventHubs/tests/Infrastructure/EventHubScope.cs]: Removed all comments to fit track one styles, and make this internal like TestUtility.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTestBase.cs
Comment (tzhanl) [reply sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTestBase.cs]: Yes, this not really use Azure service. Undo the change and remove [LiveTest] attribute.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs]: We don't need 4 partitions for this test.   Looks like 1 would be ideal.    Can we please review the partition counts that we're using to ensure that we're only requesting the number of partitions that a test uses, or that are needed to validate the scenario?

Comment (jsquire) [reply sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs]: Leaving one comment here, but I'd appreciate if we could do that diligence across all of the tests.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/WebSocketTests.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/WebSocketTests.cs]: nit:  extra blank lines that we should remove.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Properties/AssemblyInfo.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Properties/AssemblyInfo.cs]: Let's go ahead and remove this, please.

**Review by jsquire — COMMENTED** [review]
Looks like we're pretty close;  I'd like to do a pass over the number of partitions that each test is using and ensure that we're not creating unnecessary resources.

I also want to ask Serkant from the Event Hubs service team for his review before we unblock and merge.

**Comment by jsquire** [comment]
@serkantkaraca:  Before we move forward with the merge, would you be so kind as to give this a quick pass to see if anything jumps out at you?


//cc: @AlexGhiondea

**Comment by serkantkaraca** [comment]
> @serkantkaraca: Before we move forward with the merge, would you be so kind as to give this a quick pass to see if anything jumps out at you?
> 
> //cc: @AlexGhiondea

Sure, I will do today.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs
Comment (tzhanl) [reply sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs]: For some tests would access partition 1 or 2, we create 2 or 3 partitions. Exception this case, all other tests updated to only use one 1 partition.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/WebSocketTests.cs
Comment (tzhanl) [reply sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/WebSocketTests.cs]: Removed extra blank for all files we updated.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/ConnectionStringBuilderTests.cs
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/ConnectionStringBuilderTests.cs]: nit:  extra spaces

**Review by jsquire — APPROVED** [review]
Looks good to me.   I think we need to ensure we're rebased to master, but are otherwise good to move forward.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs
Comment (serkantkaraca) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs]: >ehClient.CreatePartitionSender("0"); [](start = 38, length = 36)

I think you can randomize the partition picked.

**Review by serkantkaraca — COMMENTED** [review]
Overall looks good. Approved.

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs
Comment (AlexGhiondea) [reply sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs]: @tzhanl let's create an issue to track this additional improvement!

Comment (tzhanl) [reply sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/DataBatchTests.cs]: created issue https://github.com/Azure/azure-sdk-for-net/issues/7178 to follow this.

**Pull request merged**
